pkg/interop/convert/convert.go

```diff
@@ -20,3 +25,139 @@ func ToString(v any) string {
 func ToBool(v any) bool {
 	return v.(bool)
 }
+
+// FromUintToFixedBEBytes converts an uint64 to a slice of a specified length
+// in big-endian format.
+func FromUintToFixedBEBytes(m uint64, width uint64) []byte {
+	var n = m
+	if width == 0 {
+		return []byte{}
+	}
+	var res = make([]byte, width)
+	for i := int64(width - 1); i >= 0; i-- {
+		res[i] = byte(n & 0xff)
+		n = n >> 8
+	}
+	return res
+}
+
+// FromUint64ToBEBytes converts an uint64 to a fixed-length slice in big-endian format.
+func FromUint64ToBEBytes(n uint64) []byte {
+	return FromUintToFixedBEBytes(n, 8)
+}
+
+// FromUint32ToBEBytes converts an uint32 to a fixed-length slice in big-endian format.
+func FromUint32ToBEBytes(n uint32) []byte {
+	return FromUintToFixedBEBytes(uint64(n), 4)
+}
+
+// FromUint16ToBEBytes converts an uint16 to a fixed-length slice in big-endian format.
+func FromUint16ToBEBytes(n uint16) []byte {
+	return FromUintToFixedBEBytes(uint64(n), 2)
+}
+
+// FromUint8ToBEBytes converts an uint8 to a fixed-length slice in big-endian format.
+func FromUint8ToBEBytes(n uint8) []byte {
+	return FromUintToFixedBEBytes(uint64(n), 1)
+}
+
+// FromUintToBEBytes converts an uint to a fixed-length slice in big-endian format.
+func FromUintToBEBytes(n uint) []byte {
+	return FromUintToFixedBEBytes(uint64(n), uintSizeInBytes)
+}
+
+// FromIntToFixedBEBytes converts an int64 to a slice of a specified length
+// in big-endian format.
+func FromIntToFixedBEBytes(m int64, width uint64) []byte {
+	var n = m
+	if width == 0 {
+		return []byte{}
+	}
+	n = n + (1 << (width*8 - 1))
+	return FromUintToFixedBEBytes(uint64(n), width)
+}
+
+// FromInt64ToBEBytes converts an int64 to a fixed-length slice in big-endian format.
+func FromInt64ToBEBytes(n int64) []byte {
+	return FromIntToFixedBEBytes(n, 8)
+}
+
+// FromInt32ToBEBytes converts an int32 to a fixed-length slice in big-endian format.
+func FromInt32ToBEBytes(n int32) []byte {
+	return FromIntToFixedBEBytes(int64(n), 4)
+}
+
+// FromInt16ToBEBytes converts an int16 to a fixed-length slice in big-endian format.
+func FromInt16ToBEBytes(n int16) []byte {
+	return FromIntToFixedBEBytes(int64(n), 2)
+}
+
+// FromInt8ToBEBytes converts an int8 to a fixed-length slice in big-endian format.
+func FromInt8ToBEBytes(n int8) []byte {
+	return FromIntToFixedBEBytes(int64(n), 1)
+}
+
+// FromIntToBEBytes converts an int to a fixed-length slice in big-endian format.
+func FromIntToBEBytes(n int) []byte {
+	return FromIntToFixedBEBytes(int64(n), uintSizeInBytes)
+}
+
+// FromBEBytesToUint64 converts data in big-endian format to an uint64.
+func FromBEBytesToUint64(data []byte) uint64 {
+	var n uint64
+	//nolint:all // integer range not supported
```

Comment:
Disable specific linter; properly format the comment, see other usages of `nolint` in the project as an example.